### PR TITLE
add ensemble evaluation script

### PR DESCRIPTION
**Pull request by kdutia**
running this:

```
uv run scripts/ensemble/plot_ensemble_metrics.py --ensemble-name 3fphaheq --wikibase-id Q1016
```

produces a plot that shows you uncertainty metrics against f1 score, and the amount of labelling you'd skip ('referral rate') if you sent everything below that threshold to a model

image

also produces a text file of stats that looks like this, and a CSV of ensemble majority vote predictions against ground truth labels at passage-level (it had to be assembled anyway, and i thought it was an interesting artifact! can delete as well, though)

```
Ensemble name: 3fphaheq
Concept: Q1016
Number of classifiers: 10
Total predictions: 257
Run details: 
  - autumn-music-54 (a46tkrz2): finished
  - still-sea-55 (vzi3914a): finished
  - fluent-waterfall-56 (ho3faan2): finished
  - giddy-butterfly-57 (qmgxfspw): finished
  - faithful-glitter-58 (py3b57wx): finished
  - eager-sun-59 (ch5v3f24): finished
  - curious-dawn-60 (64pt0ca0): finished
  - dandy-cloud-61 (ttvifug7): finished
  - polar-wildflower-62 (vru82ct8): finished
  - neat-night-63 (0p7ilczh): finished

```

### CODE REVIEW

File: scripts/ensemble/plot_ensemble_metrics.py [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+        if tp + fn == 0:  # No positive ground truth
+            recall = 0
+        else:
+            recall = tp / (tp + fn)
+
+        if precision + recall == 0:
+            f1 = 0
+        else:
+            f1 = 2 * (precision * recall) / (precision + recall)
```
Comment (jesse-c) [inline scripts/ensemble/plot_ensemble_metrics.py:366]: **thought:** Could you use an existing function to avoid any bugs? E.g. `from sklearn.metrics import f1_score`.

File: scripts/ensemble/plot_ensemble_metrics.py [hunk 2]
```diff
@@ -0,0 +1,489 @@
+import asyncio
+import tempfile
+from pathlib import Path
+from typing import Annotated
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
```
Comment (olaughter) [inline scripts/ensemble/plot_ensemble_metrics.py:8]: Turns out we are actually implicitly importing pandas via the sdk. It would be good to pin the version explicitly in the `pyproject.toml`, kinda surprising as we use it already in a handful of places

**Review by olaughter — APPROVED** [review]
